lib/elixir/lib/system.ex

```diff
@@ -376,6 +376,36 @@ defmodule System do
   end
 
   @doc """
+  Environment variable value.
+
+  Returns the value of the environment variable
+  `varname` as a binary, or :error if the environment
+  variable is undefined.
+  """
+  @spec fetch_env(binary) :: binary | :error
+  def fetch_env(varname) when is_binary(varname) do
+    case :os.getenv(String.to_charlist(varname)) do
+      false -> :error
+      other -> List.to_string(other)
+    end
+  end
+
+  @doc """
+  Environment variable value.
+
+  Returns the value of the environment variable
+  `varname` as a binary, or raise an error if the environment
+  variable is undefined.
+  """
+  @spec fetch_env!(binary) :: binary
```

Comment:
`get_env/1` specifies this spec:
`@spec get_env(binary) :: binary | nil`

Is this incorrect also? Should it also be `@spec get_env(String.t) :: String.t | nil`